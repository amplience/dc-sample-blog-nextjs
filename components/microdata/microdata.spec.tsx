import React from 'react';
import Microdata from './microdata';
import ShallowRender from 'react-test-renderer/shallow';

describe('Microdata', (): void => {
  beforeEach(() => {
    delete process.env.URL;
  });
  it('should render a full Microdata component and BASE_URL is not defined', (): void => {
    const props = {
      headline: 'test-headline',
      description: 'description',
      imageUrl: '//test-imageurl/image-test.jpg',
      authors: [{ name: 'test-author-1' }],
      datePublished: '2019-07-07',
      dateModified: '2019-07-07'
    };
    const renderer = ShallowRender.createRenderer();
    renderer.render(<Microdata {...props} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('should render a full Microdata component and BASE_URL is defined', (): void => {
    process.env.URL = 'http://example.com/';
    const props = {
      headline: 'test-headline',
      description: 'description',
      imageUrl: '//test-imageurl/image-test.jpg',
      authors: [{ name: 'test-author-1' }],
      datePublished: '2019-07-07',
      dateModified: '2019-07-07'
    };
    const renderer = ShallowRender.createRenderer();
    renderer.render(<Microdata {...props} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should render Microdata component when author is not defined', (): void => {
    process.env.URL = 'http://example.com/';
    const props = {
      headline: 'test-headline',
      description: 'description',
      imageUrl: '//test-imageurl/image-test.jpg',
      authors: [{ name: 'test-author-1' }],
      datePublished: '2019-07-07',
      dateModified: '2019-07-07'
    };
    delete props.authors;
    const renderer = ShallowRender.createRenderer();
    renderer.render(<Microdata {...props} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
