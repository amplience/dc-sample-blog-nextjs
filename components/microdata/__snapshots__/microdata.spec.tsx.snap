// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Microdata should render Microdata component when author is not defined 1`] = `
<ArticleJsonLd
  authorName=""
  dateModified="2019-07-07"
  datePublished="2019-07-07"
  description="description"
  images={
    Array [
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=16:9",
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=4:3",
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=1:1",
    ]
  }
  publisherLogo="http://example.com//static/images/logo-100h.png"
  publisherName="Amplience"
  title="test-headline"
  url="http://example.com/mocked-path"
/>
`;

exports[`Microdata should render a full Microdata component and BASE_URL is defined 1`] = `
<ArticleJsonLd
  authorName="test-author-1"
  dateModified="2019-07-07"
  datePublished="2019-07-07"
  description="description"
  images={
    Array [
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=16:9",
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=4:3",
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=1:1",
    ]
  }
  publisherLogo="http://example.com//static/images/logo-100h.png"
  publisherName="Amplience"
  title="test-headline"
  url="http://example.com/mocked-path"
/>
`;

exports[`Microdata should render a full Microdata component and BASE_URL is not defined 1`] = `
<ArticleJsonLd
  authorName="test-author-1"
  dateModified="2019-07-07"
  datePublished="2019-07-07"
  description="description"
  images={
    Array [
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=16:9",
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=4:3",
      "//test-imageurl/image-test.jpg?w=1200&sm=aspect&aspect=1:1",
    ]
  }
  publisherLogo="/static/images/logo-100h.png"
  publisherName="Amplience"
  title="test-headline"
  url="mocked-path"
/>
`;
